src/main/java/com/boot/product/repository/ProductRepository.java

```diff
@@ -21,6 +21,8 @@ public interface ProductRepository extends JpaRepository<Product, Long> {
 
     Product findBySlug(String slug);
 
+    List<Product> findByNameContainingIgnoreCase(String name);
```

Comment:
this query method returns all product data. you should create one that retrieves only the product name

Reply:
not shure how to do this, cand you please link some reference for it?

Reply:
you can use the query annotation
@Query("select name from Product where name like '%:names%'")